From a8647197638529b8452b037d492646359aeda71d Mon Sep 17 00:00:00 2001
From: Christian Kellner <christian@kellner.me>
Date: Mon, 25 Sep 2017 23:41:20 +0900
Subject: [PATCH] Fix ARM build: remove explicit tmpl instantiation

For reasons lost to history some templates were explicitly
instantiated, most likely to fix some windows build issues.
This seems to be hopefully not necessary anymore, becasue it
breaks builds on ARM. It is also violating the C++ std (14.7
clause 5):
  "an explicit instantiation definition shall appear at
   most once in a program"
---
 include/nix/base/types.hpp | 15 ---------------
 1 file changed, 15 deletions(-)

diff --git a/include/nix/base/types.hpp b/include/nix/base/types.hpp
index 61dc2b3d..fcba767a 100644
--- a/include/nix/base/types.hpp
+++ b/include/nix/base/types.hpp
@@ -22,19 +22,4 @@
 #include <nix/base/IMultiTag.hpp>
 #include <nix/base/IDataArray.hpp>
 
-namespace nix {
-namespace base {
-
-template class Entity<base::IProperty>;
-template class Entity<base::IFeature>;
-template class NamedEntity<base::ISection>;
-template class EntityWithMetadata<base::ISource>;
-template class EntityWithMetadata<base::IBlock>;
-template class EntityWithSources<base::ITag>;
-template class EntityWithSources<base::IMultiTag>;
-template class EntityWithSources<base::IDataArray>;
-
-}
-}
-
 #endif
-- 
2.13.5

